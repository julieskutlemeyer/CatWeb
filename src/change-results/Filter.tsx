import React, { useState } from 'react'
import ButtonToggle from 'react-bootstrap/Button'
import Col from 'react-bootstrap/Col';

const types = ['Both', 'Male', 'Female'];

function Filter() {

    // Code from https://react.school/ui/button
    const [active, setActive] = useState(types[0]);
    return (
        <Col className="filter-col">
            {types.map(type => (
                <ButtonToggle
                    key={type}
                    active={active === type}
                    onClick={() => setActive(type)}
                    variant="primary"
                >
                    {type}
                </ButtonToggle>
            ))}
        </Col>
    );
}


export default Filter;