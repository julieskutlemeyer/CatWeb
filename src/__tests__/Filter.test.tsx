import React from 'react';
import renderer from 'react-test-renderer';
import Filter from '../change-results/Filter';

test('renders filter buttons', () => {
    const component = renderer.create(
        <Filter/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
