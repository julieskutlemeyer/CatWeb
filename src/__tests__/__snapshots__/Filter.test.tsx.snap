// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that filter-buttong changes states accordingly should render with given state from Redux store 1`] = `
<div
  className="filter-col col"
>
  <button
    className="btn active btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Both
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Male
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Female
  </button>
</div>
`;
